libraries/protocol/custom_authority.cpp

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright (c) 2018 Abit More, and contributors.
+ *
+ * The MIT License
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+#include <graphene/protocol/custom_authority.hpp>
+#include <graphene/protocol/operations.hpp>
+#include <graphene/protocol/restriction_predicate.hpp>
+
+#include <fc/io/raw.hpp>
+
+namespace graphene { namespace protocol {
+
+share_type custom_authority_create_operation::calculate_fee(const fee_parameters_type& k)const {
+   share_type core_fee_required = k.basic_fee;
+   core_fee_required += k.price_per_byte * (fc::raw::pack_size(restrictions) + fc::raw::pack_size(auth));
+   return core_fee_required;
+}
+
+void custom_authority_create_operation::validate()const {
+   FC_ASSERT(fee.amount >= 0, "Fee amount can not be negative");
+
+   FC_ASSERT(account != GRAPHENE_TEMP_ACCOUNT
+             && account != GRAPHENE_COMMITTEE_ACCOUNT
+             && account != GRAPHENE_WITNESS_ACCOUNT
+             && account != GRAPHENE_RELAXED_COMMITTEE_ACCOUNT,
+             "Can not create custom authority for special accounts");
```

Comment:
It is proposed to relax this restriction such that the witness and committee accounts can set custom authorities. Do any argue I should keep this restriction as is?

Reply:
I'd rather keep them. IMHO the witness and committee accounts setting custom authorities would lead to a mess in chain governance.

Reply:
IIRC we briefly discussed if custom authorities could be used to hand over bitasset management to a separate entity. IMO the solution proposed in BSIP-83 is much cleaner and more powerful as well.

Reply:
Alright, so it sounds like the consensus is to keep the restriction as is. Any dissenters, speak now or forever hold your peace. :)